lib/services/registered_addons.pm

```diff
@@ -64,20 +64,26 @@ sub check_upgraded_addons {
 }
 
 sub check_suseconnect {
-    my $output = script_output("SUSEConnect -s");
-    my $json   = Mojo::JSON::decode_json($output);
-    foreach (@$json) {
-        my $iden   = $_->{identifier};
-        my $status = $_->{status};
-        die "$iden register status is: $status" if ($status ne 'Registered');
+    my $output = script_output("SUSEConnect -s", 120);
+    my @out    = grep { $_ =~ /identifier/ } split(/\n/, $output);
+    if (@out) {
+        my $json = Mojo::JSON::decode_json($out[0]);
+        foreach (@$json) {
+            my $iden   = $_->{identifier};
+            my $status = $_->{status};
+            die "$iden register status is: $status" if ($status ne 'Registered');
+        }
+    }
+    else {
+        die "Cannot get register status: $output";
     }
 }
 
 sub full_registered_check {
     my ($stage) = @_;
     $stage //= '';
-    check_suseconnect();
     zypper_lr();
+    check_suseconnect();
```

Comment:
this does not look related

Reply:
We want to check register addons and register status. As there is one bug: https://bugzilla.suse.com/show_bug.cgi?id=1149719
Which found by customer at first. So, we want to check 'SUSEConnect -s ' before and after migration.

Reply:
Yes, sure. But this is not related to the goal of the single git commit

Reply:
Change back as it is not related :)